packages/reality-data/src/components/Widget.tsx

```diff
@@ -0,0 +1,975 @@
+    this.setState((_prevState) => {
+      return {
+        filter: value,
+      };
+    });
+  };
+
+  private _onShowMapTypes = (event: any) => {
+    event.stopPropagation();
+    const view = this._vp!.view as SpatialViewState;
+    const mapType = view.getDisplayStyle3d().backgroundMap.settings.mapType;
+    const showMapTypes = !this.state.showMapTypes;
+    this.setState({
+      showMapTypes,
+      bingMapType: mapType,
+    });
+  };
+
+  private _onChanged = (mapType: BackgroundMapType) => {
+    if (!this._vp) return;
+
+    this._vp!.changeBackgroundMapProps({ providerData: { mapType } });
+    this._vp!.synchWithView(false);
+
+    this._vp!.invalidateScene();
+    this.setState({ showMapTypes: false, bingMapType: mapType });
+  };
+
+  private _onSettingsOpened = () => {
+    this.fullAppContext.trackEvent({ name: "backgroundMapSettings" });
+    this.setState({ isSettingsOpen: true, showMapTypes: false });
+  };
+
+  private _onCloseMapTypesMenu = () => {
+    this.setState({ showMapTypes: false });
+  };
+
+  private _isMapVisible = () => {
+    const mapLabel = RealityData.translate("bingMap");
+    return (
+      this.state.filter.length === 0 ||
+      regexFromSearchQuery(this.state.filter).test(mapLabel)
+    );
+  };
+
+  private _toggleBingMap = () => {
+    const view = this._vp!.view as SpatialViewState;
+    const newFlags = view.viewFlags.clone();
+    newFlags.backgroundMap = !view.viewFlags.backgroundMap;
+    this._vp!.viewFlags = newFlags;
+    const isMapEnabled = this._vp!.viewFlags.backgroundMap;
+
+    if (isMapEnabled) {
+      if (
+        view.getDisplayStyle3d().settings.backgroundMap.providerName !==
+        "BingProvider"
+      ) {
+        this._vp!.changeBackgroundMapProps({
+          providerName: "BingProvider",
+          providerData: { mapType: this.state.bingMapType },
+        });
+      }
+
+      this._vp!.synchWithView(false);
+    }
+
+    this.setState(
+      { isMapEnabled },
+      () => this._vp && this._vp.invalidateScene()
+    );
+  };
+
+  private _getFilteredRealityData = (): Entry[] => {
+    if (!this.state.filter || this.state.filter === "") return this.state.items;
+    const exp = regexFromSearchQuery(this.state.filter);
+
+    let filteredData: Entry[] = [];
+    if (this.state.items) {
+      filteredData = this.state.items!.filter((item) => {
+        return exp.test(item.name) || exp.test(item.description);
+      });
+    }
+    return filteredData;
+  };
+}
+
+/** Reality Data Widget Control component */
+export class Control extends WidgetControl {
+  constructor(info: ConfigurableCreateInfo, options: WidgetProps) {
+    super(info, options);
+    this.reactElement = (
+      <div
+        style={{
+          padding: "2px 4px 0px 4px",
+          width: "100%",
+          height: "100%",
+          boxSizing: "border-box",
+        }}
+      >
+        <Widget {...options} />
+      </div>
+    );
+  }
+}
+
+export default Control;
```

Comment:
should prefer named exports imo. just keep it consistent through out this package